pkg/operator/pool/controller.go

```diff
@@ -197,12 +198,17 @@ func ValidatePoolSpec(context *clusterd.Context, namespace string, p *rookalpha.
 		return fmt.Errorf("neither replication nor erasure code settings were specified")
 	}
 
-	// validate the failure domain if specified
-	if p.FailureDomain != "" {
-		crush, err := ceph.GetCrushMap(context, namespace)
+	var crush ceph.CrushMap
+	var err error
+	if p.FailureDomain != "" || p.CrushRoot != "" {
```

Comment:
a `p.CrushRoot != ""` implies that the `default` CrushRoot is to be used. So when the FailureDomain is not specified by we are to use the `default` crush rule, you dont need to get the crush map?

Reply:
we get the crush map so we can verify if the `FailureDomain` and/or `CrushRoot` exist. If they don't exist, the validation will fail. But if those two properties are not set, we will use the default values, which always exist.